Sources/Mail.php

```diff
@@ -208,52 +209,16 @@ public static function send($to, $subject, $message, $from = null, $message_id =
 			Config::updateModSettings(array('mail_recent' => $new_queue_stat));
 		}
 
-		// SMTP or sendmail?
-		if ($use_sendmail)
-		{
-			$subject = strtr($subject, array("\r" => '', "\n" => ''));
-			if (!empty(Config::$modSettings['mail_strip_carriage']))
-			{
-				$message = strtr($message, array("\r" => ''));
-				$headers = strtr($headers, array("\r" => ''));
-			}
-
-			foreach ($to_array as $to)
-			{
-				set_error_handler(
-					function($errno, $errstr, $errfile, $errline)
-					{
-						// error was suppressed with the @-operator
-						if (0 === error_reporting())
-							return false;
+		// Loadup the agent.
+		$agent = MailAgent::load();
+		if ($agent === false || !$agent->connect())
+			return false;
 
-						throw new ErrorException($errstr, 0, $errno, $errfile, $errline);
-					}
-				);
-				try
-				{
-					if (!mail(strtr($to, array("\r" => '', "\n" => '')), $subject, $message, $headers))
-					{
-						ErrorHandler::log(sprintf(Lang::$txt['mail_send_unable'], $to));
-						$mail_result = false;
-					}
-				}
-				catch (ErrorException $e)
-				{
-					ErrorHandler::log($e->getMessage(), 'general', $e->getFile(), $e->getLine());
-					ErrorHandler::log(sprintf(Lang::$txt['mail_send_unable'], $to));
-					$mail_result = false;
-				}
-				restore_error_handler();
+		$mail_result = true;
+		foreach ($to_array as $to)
+			$mail_result &= $agent->send($to, $subject, $message, $headers);
```

Comment:
let's  keep it a bool
```suggestion
			$mail_result = $mail_result && $agent->send($to, $subject, $message, $headers);
```
I wish `&&=`  was in php

Reply:
Any reason to use that over the &= bitwise operator?

Reply:
strictness; no type juggling

Reply:
Enlighten me, where did I juggle?  $mail_result is a bool and send returns a bool.

Reply:
Right I see it now.  &= is doing a mathematical operation and adding, not as you mentioned it needs, &&=